src/MLJNearestNeighborsInterface.jl

```diff
@@ -1,36 +1,129 @@
+export DualU, DualD, Dudani, Fibonacci, Inverse, ISquared, KNNKernel, Macleod, Rank, ReciprocalRank,
+    UDK, Uniform , Zavreal
+     
 
-const srcdir = dirname(@__FILE__) # the directory containing this file
+# ===================================================================
+## CONSTANTS
+const Vec{T} = AbstractVector{T}
+const Mat{T} = AbstractMatrix{T}
+const Arr{T, N} = AbstractArray{T, N}
+const ColumnTable =  Tables.ColumnTable
+const DictTable = Dict{Symbol, <:AbstractVector}
+const MultiUnivariateFinite = Union{DictTable, ColumnTable}
 
-# TODO remove when the functionality has been merged in ScientificTypes.jl
-# and use ScientificTypes.nonmissing then.
-if VERSION < v"1.3"
-    nonmissingtype(::Type{T}) where T =
-        T isa Union ? ifelse(T.a == Missing, T.b, T.a) : T
-end
+# Define constants for easy referencing of packages
+const MMI = MLJModelInterface
+const NN = NearestNeighbors
+const PKG = "MLJNearestNeighborsInterface"
 
+# Definitions of model descriptions for use in model doc-strings.
+const KNNRegressorDescription = """
+    K-Nearest Neighbors regressor: predicts the response associated with a new point
+    by taking an weighted average of the response of the K-nearest points.
+    """
 
-# # lazily load in strap-on model interfaces for external packages:
-# function __init__()
+const KNNClassifierDescription = """
+    K-Nearest Neighbors classifier: predicts the class associated with a new point
+    by taking a vote over the classes of the K-nearest points.
+    """
 
-    @require(NearestNeighbors="b8a86587-4115-5ab1-83bc-aa920d37bbce",
-             include("NearestNeighbors_.jl"))
+const KNNCoreFields = """
+    * `K::Int=5` : number of neighbors
+    * `algorithm::Symbol = :kdtree` : one of `(:kdtree, :brutetree, :balltree)`
+    * `metric::Metric = Euclidean()` : a `Metric` object for the distance between points
+    * `leafsize::Int = 10` : at what number of points to stop splitting the tree
+    * `reorder::Bool = true` : if true puts points close in distance close in memory
+    * `weights::KNNKernel=Uniform()` : kernel used in assigning weights to the 
+        k-nearest neighbors for each observation. An instance of one of the types in 
+        `list_kernels()`. User-defined weighting functions can be passed by wrapping the 
+        function in a `UDF` kernel. If sample weights `w` are passed during machine 
+        construction e.g `machine(model, X, y, w)` then the weight assigned to each 
+        neighbor is the product of the `KNNKernel` generated weight and the corresponding 
+        neighbor sample weight.
+     
+    """
 
-# end
+const SeeAlso = """
+    See also the 
+    [package documentation](https://github.com/KristofferC/NearestNeighbors.jl).
+    For more information about the kernels see the paper by Geler et.al 
+    [Comparison of different weighting schemes for the kNN classifier 
+    on time-series data](https://perun.pmf.uns.ac.rs/radovanovic/publications/
```

Comment:
It seems there is a truant space between `/publications/` and `2016` when I query the docstring in the REPL, leading to link not working.

Reply:
true